src/components/form/carForm.tsx

```diff
@@ -1,115 +1,115 @@
+                        label='Year'
+                        error={errors.year !== undefined}
+                        helperText={errors.year?.message?.toString()}
+                        fullWidth
+                        {...field}
+                    />
+                )}
+            />
+            <Controller
+                name='engine'
+                control={control}
+                rules={{required: 'Engine Type is required'}}
+                render={({field}) => (
+                    <FormControl error={errors.engine !== undefined} required component="fieldset">
+                        <FormLabel component="legend">Engine Type</FormLabel>
+                        <RadioGroup 
+                            row
+                            aria-label="engine" 
+                            {...field}>
+                            {engineTypes.map((engine) => (<FormControlLabel value={engine} control={<Radio />} label={engine} />))}
+                        </RadioGroup>
+                        <FormHelperText>{errors.engine?.message?.toString()}</FormHelperText>
+                    </FormControl>
+                )}
             />
-            <FormControl required component="fieldset" error={Boolean(errors.engine)}>
-                <FormLabel component="legend">Engine Type</FormLabel>
-                <RadioGroup 
-                    row
-                    aria-label="engine" 
-                    name="engine" 
-                    value={car.engine} 
-                    onChange={handleChange}>
-                    {engineTypes.map((engine) => (<FormControlLabel value={engine} control={<Radio />} label={engine} />))}
-                </RadioGroup>
-                <FormHelperText>{errors.engine}</FormHelperText>
-            </FormControl>
         </>
     )
 }
 
 export default function CarForm({ addCar } : CarFormProps){
-    const [car, setCar] = useState(emptyCar) // Car object state
-    const [errors, setErrors] = useState<FormErrors>(emptyCar) // Errors state
-
-    useEffect(() => { // Effect to re-validate form fields
-        if (Object.values(errors).some((value) => value !== '')){
-            validateCar({car, errors, setErrors})
-        }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [car])
-    
-    const handleChange = (e: ChangeEvent<HTMLInputElement>) => { // Handler for input changes
-        const { name, value } = e.target
-        setCar({ ...car, [name]: value })
-    }
-    
-    const handleSubmit = (e: FormEvent) => { // Handler for form submission
-        e.preventDefault()
-        if (!validateCar({car, errors, setErrors})) return
-        addCar(car)
-        setCar(emptyCar)
-    }
 
-    const handleReset = () => { // Handler for form reset
-        setCar(emptyCar)
-        setErrors(emptyCar)
-    }
+    const { control, handleSubmit: submit, formState, reset } = useForm<Car>()
+    const onSubmit: SubmitHandler<Car> = (data) => addCar(data)
```

Comment:
You can set in default form for a values when in the useForm hook:
` useForm<Car>({defaultValues: emptyCar})`